File: server/php/src/Application/Interfaces/SDKClients/PaymentClientInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\SDKClients;

use OnlinePayments\ExampleApp\Application\DTOs\Payment\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\ResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\AdditionalPaymentActions\RequestDto as AdditionalPaymentActionsRequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\AdditionalPaymentActions\ResponseDto as AdditionalPaymentActionsResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\GetPaymentDetails\ResponseDto as GetPaymentDetailsResponseDto;

interface PaymentClientInterface
{
    public function createPayment(RequestDto $requestDto): ResponseDto;
    public function getPaymentDetailsById(string $id): ?GetPaymentDetailsResponseDto;
    public function cancelPayment(AdditionalPaymentActionsRequestDto $requestDto): AdditionalPaymentActionsResponseDto;
    public function capturePayment(AdditionalPaymentActionsRequestDto $requestDto): AdditionalPaymentActionsResponseDto;
    public function refundPayment(AdditionalPaymentActionsRequestDto $requestDto): AdditionalPaymentActionsResponseDto;
}

File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/clients/PaymentClientInterface.java
package com.onlinepayments.example.service.interfaces.clients;

import com.onlinepayments.example.service.dtos.additionalpaymentactions.AdditionalPaymentActionRequestDto;
import com.onlinepayments.example.service.dtos.additionalpaymentactions.AdditionalPaymentActionResponseDto;
import com.onlinepayments.example.service.dtos.paymentdetails.PaymentDetailsResponseDto;
import com.onlinepayments.example.service.dtos.payments.CreatePaymentRequestDto;
import com.onlinepayments.example.service.dtos.payments.CreatePaymentResponseDto;

public interface PaymentClientInterface {
    CreatePaymentResponseDto createPayment(CreatePaymentRequestDto request);

    AdditionalPaymentActionResponseDto capturePayment(AdditionalPaymentActionRequestDto request);

    AdditionalPaymentActionResponseDto refundPayment(AdditionalPaymentActionRequestDto request);

    AdditionalPaymentActionResponseDto cancelPayment(AdditionalPaymentActionRequestDto request);

    PaymentDetailsResponseDto getPaymentDetailsById(String id);
}
